### ISSUE TITLE
ConvQuadInterp3d float16 backward returns NaN on CPU (Cramer solve division)

### ISSUE THREAD

**Issue by ducha-aiki** [kornia/kornia#4257]
## Summary

`ConvQuadInterp3d` / `conv_quad_interp3d` return finite coordinates and values in float16 on CPU, but the gradient with respect to the input contains NaN over part of the volume. bfloat16 and float32 are clean. Present on `main` (`1eea5835`); #4254 does not touch `kornia/geometry/subpix/` and reproduces it identically.

## Reproduction

```python
import torch
from kornia.geometry.subpix import ConvQuadInterp3d

torch.manual_seed(0)
for dt in (torch.float16, torch.bfloat16, torch.float32):
    x = torch.rand(1, 1, 5, 40, 40, dtype=dt, requires_grad=True)
    coords, vals = ConvQuadInterp3d(strict_maxima_bonus=0.0)(x)
    g = torch.autograd.grad(coords.sum() + vals.sum(), x)[0]
    print(dt, coords.isnan().any().item(), vals.isnan().any().item(), g.isnan().sum().item(), "/", g.numel())
```

torch 2.14.0, CPU, macOS arm64:

```
torch.float16 False False 614 / 8000
torch.bfloat16 False False 0 / 8000
torch.float32 False False 0 / 8000
```

`torch.autograd.detect_anomaly()` names `DivBackward0` from `_solve_cramer_sym3x3` (`kornia/geometry/subpix/spatial_soft_argmax.py`, the `/ safe_det` divisions that produce `sx`, `sy`, `ss`), reached through `conv_quad_interp3d`.

Through `ScaleSpaceDetector(minima_are_also_good=True)` with `BlobDoG` on a 96×99 float16 image, the input gradient has 2161 NaN entries of 9504. The count is identical on `main` and on #4254, on both the stacked and the separate refinement path.

## Likely mechanism, not yet verified

The determinant of the 3×3 Hessian is a product of three second derivatives and sits around 1e-6 to 1e-9 for a [0, 1] image, in or below float16's subnormal range. `solved = det.abs() > eps` admits those entries, and the backward of `num / safe_det` multiplies by `1 / safe_det**2`, which overflows to inf in float16 and becomes NaN in the reduction. The forward divides once and stays finite. The default `eps=1e-7` of `_solve_cramer_sym3x3` is below `torch.finfo(torch.float16).tiny` (6.1e-5), so the threshold admits subnormal determinants in float16. A check is to compare `det.abs()` at the NaN positions against `torch.finfo(torch.float16).tiny`.

## Notes

- `tests/feature/test_scale_space_detector.py::TestScaleSpaceDetector::test_joint_extrema_matches_separate_refinement` compares gradients in every dtype but float16 for this reason (#4254); once fixed, that guard can go.
- Whether CUDA float16 is affected is unverified (no CUDA here).

Posted on behalf of @ducha-aiki by Claude (Fable 5.1).
